Custom legends can be specified for filter aggregates

K-Phoen opened the pull request:
Closes #2245

This PR allows the user to define custom labels in filter aggregates, as shown:

![this screenshot](https://cloud.githubusercontent.com/assets/66958/7197390/d2d055ba-e4e0-11e4-9a5e-c826cf6b5b98.png).

It's still WIP and this should also be implemented for range aggregates but it already works.

Let me know what you think :)

rashidkpc commented on src/kibana/components/agg_types/buckets/filters.js:
```diff
@@ -29,7 +29,7 @@ define(function (require) {
               decorateQuery(query);
 
-              var label = _.deepGet(query, 'query_string.query') || angular.toJson(query);
+              var label = filter.label ? filter.label : (_.deepGet(query, 'query_string.query') || angular.toJson(query));
```
Can you not use a ternary here? Make this hard to read.

K-Phoen replied:
Done.

rashidkpc commented:
This is a good start, something seems off about the label input box, spacing or something. Also, when you have a long label it interferes with the buttons:

![image](https://cloud.githubusercontent.com/assets/1250387/7240629/56d37602-e767-11e4-8c84-e175cf362334.png)

K-Phoen commented:
I added a `<label>` for the legend field and fixed the "long label interfering with the buttons" issue but I didn't understand what you meant by "*something seems off about the label input box, spacing or something*".

rashidkpc commented:
The spacing of the label on the input box is a bit more than it is for the other labels, likely due to the use of `vis-editor-agg-header-title` which isn't really meant for input box labels.

![screen shot 2015-04-24 at 10 23 18 am](https://cloud.githubusercontent.com/assets/1250387/7324523/fd1fc9aa-ea6b-11e4-864e-df1085b7915d.png)

K-Phoen commented:
Should be better now.

rashidkpc commented on src/kibana/plugins/visualize/editor/styles/editor.less:
```diff
@@ -153,6 +153,14 @@
         font-weight: bold;
       }
 
+      &-growing {
```
Not sure on this class naming? Also, this really isn't any kind of `vis-editor-agg-header`

K-Phoen commented:
I updated the CSS definition.

rashidkpc commented:
Looks good, I expect to merge this right after we cut 4.1

K-Phoen commented:
Great, thanks for your patience :)

elasticsearch-release commented:
Jenkins standing by to test this. If you aren't a maintainer, you can ignore this comment. Someone with commit access, please review this and clear it for Jenkins to run; then  say 'jenkins, test it'.

peterdm commented:
Saw this merge.  Even tested it out on master.  Thanks!  Extremely helpful!

However, as of today, I'm seeing commits going to master requiring Elasticsearch 2.0.0 to run Kibana.
132740d61dc05dc359899eec11c19367f3c93d69

Any plan to splice out a 4.2 Kibana release that still supports 1.4.x?

pmoust commented:
Has this been in a release?

evansendra commented:
^ +1.  is this released?  Using kibana 4.1.1 build 7489

megglos commented:
@K-Phoen hope you don't mind me creating this backport PR for 4.1, https://github.com/elastic/kibana/pull/4767
